### ISSUE TITLE
Let me set ENV vars global and per project and it should work for "everything" inside VSCode

### ISSUE THREAD

**Issue by alanwilter** [microsoft/vscode#119381]
I use Mac local and Linux via VS Code remote.

I use `Terminal` (cmd+J) and `Test` (from left side bar on VSCode, the erlenmeyer flask).

It's well known that starting with command line `code .` from the folder of a project and all ENVs, included the ones set with `export ...` are sorted. So I can open `Terminal` or run `Test` and everything will work as expected, ON LOCAL computer (I need to pass AWS credentials and a special APP_ENV var for `pytest`).

However, `Terminal` on remote linux will NOT, for **obvious reasons**. Nevertheless, one thing that puzzles me is how `Test` on remote knows everything, AWS, correct APP_ENV etc. yet it's only set in `private.env` file. And since I can find where this is set I can't change it! So if I want to change `APP_ENV=dev` for example? This is concerning about `Test` (the "Flask" icon and I use `pytest`)

Now, if I start VSCode with e.g. mac "Spotlight", only ENVs from my `~./zshrc` will be there.

So, I'm desperately wondering WHY VSCode does not seem to offer the possibility of setting whole ENVs and project ENVs in e.g. `settings.json`. Or there is such a thing and I'm just stupid for not finding it?

I can't set, for example, a way to run `Test` using my specific ENV values. Sure I can do things with `launch.json`and, certainly, I can do in `.vscode/settings.json`:
``` json
    "terminal.integrated.env.linux": {
        "APP_ENV": "prod",
...
```
But alas, it seems I have set this for every platform I use and it will only work for `Terminal`, **NOT** for `Test`!

Bottom line, I can set some ENVs, but only for `Terminal`. I can set `APP_ENV` but only for my local mac starting `VSCode` with `code .`. But I can't set `APP_ENV` on my remote `VSCode`.

**Comment by chrmarti** [comment]
Which Python extension is this?

Maybe the extension could be changed to read `"terminal.integrated.env.linux"`? @Tyriar Do we have any guidance on this?

**Comment by alanwilter** [comment]
I'm using `ms-python.python v2021.4.690665134-dev`.
As about `terminal.integrated.env.{linux,osx}`, perhaps this is a possibility.

**Comment by Tyriar** [comment]
It's not clear what this `Test` thing you're referring to is.

As for the extension, the recommended way of an extension influencing all terminal environments is via the environment variable collection API. I don't think extensions should be attempting to read the `terminal.integrated.env...` settings as that should happen automatically. This discussion probably should live in the python repo.

**Comment by chrmarti** [comment]
/causedByExtension

This is caused by the Python extension. Please file a new issue for it: https://github.com/microsoft/vscode-python/issues/new/choose

**Comment by alanwilter** [comment]
Thanks for spotting the problem. I even found more issues related, I'll report to `ms-python.python`.
